### ISSUE TITLE
Implementing foreign links on generic errors

### ISSUE THREAD

**Issue by ayoshi** [brson/error-chain#146]
Hi.

Is there any way to accomplish this?
For something like ``` std::sync::PoisonError<T> ``` for example.

**Comment by Yamakaky** [comment]
No, there is no polymorphism. You can foreign link `PoinsonError<Thing>` though for example.

**Comment by ayoshi** [comment]
Thank you, yes, that much I succeed to do.
The problem arises when there is a generic error, especially with lifetime
parameters.

I think it could be implemented with addiitional parametric macro
implementation - but I worked around it, so I wouldn't be able to give it a
try.

**Comment by jsgf** [comment]
Is there a plan to implement this? It has come up a few times for me.

**Comment by Yamakaky** [comment]
Yeah, I know... If you have an idea of implementation, it's more than welcome ^^

**Comment by Object905** [comment]
There is [parse_generics_shim](https://docs.rs/parse-generics-shim/*/parse_generics_shim/index.html), which is macro from closed RFC, that allows parsing generics and where clause in macro, it's limited, but possible solution.
